### DISCUSSION TITLE
VSC extension PR idea: codegen-on-save option (default false)

### DISCUSSION THREAD

**Discussion by metaleap** [a-h/templ#201] [Q&A]
Would you be open to this functionality idea and a PR for that? Asking before I waste time... 

Way I see that, with that new config-setting contribution set to `true`, on any Save or SaveAll action involving `.templ` files, the changed files would be regenerated. As for diagnostics, I'm not sure... 

And of course, if this idea got you all eager to hack this in yourself, lemme know and I'll let ya! :grin:

**Comment by a-h** [comment]
Hi @metaleap - sorry about not getting back to you on this one, had too many things to respond to and do and lost this.

`templ generate` now has a hot reload (`--watch`) feature which watches the files in the background and automatically regenerates templ file. This has some other features like reloading the content in the browser automatically - see https://templ.guide/commands-and-tools/hot-reload/

I think that can give you something similar to what you want - although, it does involve running a command in the terminal - it's not built into the LSP feature.

![hot_reload](https://github.com/a-h/templ/assets/1029947/86181363-0054-4418-bef0-fcdc53af6af8)

The LSP that powers the VS Code extension already regenerates templ files in memory to provide the existing diagnostics, but doesn't write them to disk. I suppose the LSP could be updated so that it writes out to disk if/when the parsed templ file is valid.

However, I do worry about clashes between programs that are updating that templ file, e.g. how it would interact with the `--watch` mode, or if someone if using `air` in the background to regenerate templ files.  For reference, the new `--watch` mode generates different code than without the flag, to load strings from a file on disk so that it doesn't need to recompile / restart the go program if a string has been modified in a template.

I don't want to throw out the idea, because it sounds very useful to generate the files automatically on change from within the editor plugin, and not have to run something separate. I suppose the `--watch` mode could have another flag to stop it from generating code, leaving it to the LSP instead.
